lib/state_machine.js

```diff
@@ -27,99 +27,196 @@ limitations under the License.
     module.exports = factory();
   }
 }(function() {
```

Comment:
are we able to use ES6 modules now to avoid this gobbledygook?  I wouldn't do that in this CL, but this weird function wrapper business made my eyes roll.